demo.js

```diff
@@ -123,6 +123,18 @@
   var defaultBackground = palettes.style.backgroundColor;
   var tooltip = byId('tooltip');
 
+  function hex(x) {
+    return ("0" + parseInt(x).toString(16)).slice(-2);
+  }
+
+  function rgbToHex(rgb) {
+    if (!rgb) {
+      return "";
+    }
+    rgb = rgb.match(/^rgba?\((\d+),\s*(\d+),\s*(\d+)(,\s*\d+\.*\d+)?\)$/);
```

Comment:
Is it guaranteed that `foo.style.backgroundColor` uses `rgb(r, g, b)` format? I would feel better if the function handled match failure:
```
var m = rgb && rgb.match(…);
return m ? … : '';
```

Reply:
Ah good idea